Yank/experiment.py

```diff
@@ -1351,6 +1363,25 @@ def _system_files(files):
                 return [filepath for (ext, filepath) in sorted(zip(provided_extensions, files))]
             return _system_files
 
+        def check_regions_clash(system_dict, mol_class1, mol_class2=None):
+            """
+            Check that the regions have non clashing names by looking at regions in each molecule for a given
+            system
+            """
+            mol_description1 = self._db.molecules[system_dict[mol_class1]]
+            mol_description2 = self._db.molecules[system_dict[mol_class2]] if mol_class2 is not None else {}
+            # Fetch the regions or an empty dict
+            regions_1_names = mol_description1.get('regions', {}).keys()
+            regions_2_names = mol_description2.get('regions', {}).keys()
+            for region_1_name in regions_1_names:
+                if region_1_name in regions_2_names:
```

Comment:
You could use the `set.intersection()` operation here to check for repeated regions names. I think just doing `set(mol_description1.get('regions', {}))` should do the trick.

Reply:
I could, but I want the error message to be helpful, although thats currently having problems since the error stack is garbage at this point.